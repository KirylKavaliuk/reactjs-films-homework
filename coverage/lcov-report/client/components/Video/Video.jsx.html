<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Video/Video.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Video</a> Video.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>12/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>15/15</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
&nbsp;
import Loading from 'components/Loading/Loading';
&nbsp;
import http from 'utils/http';
&nbsp;
import styles from './Video.scss';
&nbsp;
export default class Video extends Component {
  state = {
    src: '',
    error: false,
    loaded: false,
  }
&nbsp;
  componentDidMount() {
    this.loadVideoSrc();
  }
&nbsp;
  loadVideoSrc = async () =&gt; {
    const { id } = this.props;
&nbsp;
    try {
      const { results } = await http.get(`db/movie/${id}/videos`);
&nbsp;
      console.log(results);
&nbsp;
      if(results &amp;&amp; results[0] &amp;&amp; results[0].key) {
        this.setState({ src: `http://www.youtube.com/embed/${results[0].key}` });
      } else {
        throw new Error('not loaded');
      }
    } catch(err) {
      console.log(err);
      this.setState({ error: true, loaded: true }); 
    }
  }
&nbsp;
  onLoadHandler = () =&gt; {
    this.setState({ loaded: true });
  }
&nbsp;
  render() {
    const { src, error, loaded } = this.state;
&nbsp;
    return (
      &lt;div className={ styles.videoWrapper }&gt;
        { !error
        &amp;&amp; &lt;iframe
            onLoad={ this.onLoadHandler }
            className={ styles.video }
            src={ src }
            allowFullScreen
          &gt;&lt;/iframe&gt; }
        { !error &amp;&amp; !loaded &amp;&amp; &lt;Loading className={ styles.loading }/&gt; }
        { error &amp;&amp; &lt;div className={ styles.error }&gt;Error! Video is not loaded.&lt;/div&gt; }
      &lt;/div&gt;
    );
  }
}
&nbsp;
Video.defaultProps = {
  id: 0,
};
&nbsp;
Video.propTypes = {
  id: PropTypes.number,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
