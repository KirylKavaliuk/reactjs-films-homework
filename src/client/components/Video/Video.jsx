import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Loading from 'components/Loading/Loading';

import http from 'utils/http';

import styles from './Video.scss';

export default class Video extends Component {
  state = {
    src: '',
    error: false,
    loaded: false,
  }

  timer = null;

  componentDidMount() {
    this.timer = setTimeout(() => {
      if (!this.state.loaded) {
        this.setState({ error: true });
      }
    }, 7000);

    http.get(`db/movie/${this.props.id}/videos`)
      .then(({ results }) => {
        this.setState({
          src: `http://www.youtube.com/embed/${results[0].key}`,
        });
      });
  }

  onLoadHandler = () => {
    this.setState({ loaded: true });
  }

  componentWillUnmount() {
    clearTimeout(this.timer);
  }

  render() {
    const { src, error, loaded } = this.state;

    return (
      <div className={ styles.videoWrapper }>
        { !error
        && <iframe
            onLoad={ this.onLoadHandler }
            className={ styles.video }
            src={ src }
            allowFullScreen
          ></iframe> }
        { !error && !loaded && <Loading className={ styles.loading }/> }
        { error && <div className={ styles.error }>Error! Video is not loaded.</div> }
      </div>
    );
  }
}

Video.defaultProps = {
  id: 0,
};

Video.propTypes = {
  id: PropTypes.number,
};
