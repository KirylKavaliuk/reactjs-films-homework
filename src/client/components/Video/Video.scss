@import 'styles/palette';

.video-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  width: 70vw;
  height: 40vw;
  padding: 10px;
  position: relative;
}

.video {
  width: 100%;
  height: 100%;
  border: none;
}

.error {
  font-size: 20px;
  color: $red;
}

.loading {
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: -1;
}

@media only screen and (max-width: 700px) {
  .video-wrapper {
    width: 85vw;
    height: 50vw;
  }
}

@media only screen and (max-width: 450px) {
  .video-wrapper {
    width: 95vw;
    height: 60vw;
  }
}
