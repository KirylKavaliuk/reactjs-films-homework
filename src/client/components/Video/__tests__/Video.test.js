import React from 'react';
import renderer from 'react-test-renderer';
import Video from '../Video';

jest.useFakeTimers();

it('<Video/>', () => {
  const tree = renderer.create(<Video id={ 297802 }/>);

  jest.advanceTimersByTime(2000);

  const video = tree.toJSON();

  video.children[0].props.onLoad();

  expect(video).toMatchSnapshot();
  tree.unmount();
});

it('<Video/>', () => {
  const tree = renderer.create(<Video id={ 66666666666 }/>);

  jest.advanceTimersByTime(7777);

  const video = tree.toJSON();

  expect(video).toMatchSnapshot();
});
