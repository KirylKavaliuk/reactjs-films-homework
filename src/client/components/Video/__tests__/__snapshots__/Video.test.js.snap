// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders video with timers correctly 1`] = `
<div
  className="Video__video-wrapper___2WJzy"
>
  <iframe
    allowFullScreen={true}
    className="Video__video___2sO_i"
    onLoad={[Function]}
    src=""
  />
  <div
    className="Loading__loading___2_FrG Video__loading___1yK56"
  >
    <div
      className="Loading__icons___1uA0Q"
    >
      <div
        className="Loading__icon___2WVPJ"
      >
        <div
          className="Loading__circle___14YyA"
        />
      </div>
      <div
        className="Loading__icon___2WVPJ"
      >
        <div
          className="Loading__circle___14YyA"
        />
      </div>
      <div
        className="Loading__icon___2WVPJ"
      >
        <div
          className="Loading__circle___14YyA"
        />
      </div>
    </div>
    <span
      className="Loading__text___13r1Y"
    >
      Loading
    </span>
  </div>
</div>
`;

exports[`renders video with timers uncorrectly 1`] = `
<div
  className="Video__video-wrapper___2WJzy"
>
  <div
    className="Video__error___25AAU"
  >
    Error! Video is not loaded.
  </div>
</div>
`;
